src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -9,67 +13,511 @@ void EnKarebaba_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void EnKarebaba_Update(Actor* thisx, GlobalContext* globalCtx);
 void EnKarebaba_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
+void EnKarebaba_SetupGrow(EnKarebaba*);
+void EnKarebaba_SetupIdle(EnKarebaba*);
+void EnKarebaba_Grow(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Idle(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Awaken(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Spin(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Dying(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_DeadItemDrop(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Retract(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Dead(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Regrow(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Upright(EnKarebaba* this, GlobalContext* globalCtx);
+
 const ActorInit En_Karebaba_InitVars = {
     ACTOR_EN_KAREBABA,
     ACTORTYPE_ENEMY,
-    FLAGS,
+    5,
     OBJECT_DEKUBABA,
     sizeof(EnKarebaba),
     (ActorFunc)EnKarebaba_Init,
     (ActorFunc)EnKarebaba_Destroy,
     (ActorFunc)EnKarebaba_Update,
-    (ActorFunc)EnKarebaba_Draw,
+    (ActorFunc)EnKarebaba_Draw
+};
+
+static ColliderCylinderInit sBodyCollider = {
+    { 0xC, 0, 9, 0, 0x10, COLSHAPE_CYLINDER },
+    { 0, { 0x00000000, 0, 0 }, { ~0x00300000, 0, 0 }, 0, 1, 0 },
+    { 7, 25, 0, { 0, 0, 0 } }
+};
+
+static ColliderCylinderInit sHeadCollider = {
+    { 0x0C, 0x11, 0, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0, { ~0x00300000, 0, 8 }, { 0x00000000, 0, 0 }, 9, 0, 1 },
+    { 4, 25, 0, { 0, 0, 0 }}
+};
+
+static CollisionCheckInfoInit sKarebabaCheckInfo = {
+    1, 15, 80, 254
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(unk_4C, 0x9C4, ICHAIN_CONTINUE),
+    ICHAIN_U8(unk_1F, 1, ICHAIN_CONTINUE),
+    ICHAIN_S8(naviEnemyId, 9, ICHAIN_STOP)
+};
+
+static Vec3f sVecZero1 = { 0.0f, 0.0f, 0.0f };
+
+static Color_RGBA8 sColorBlack = { 0, 0, 0, 0 };
```

Comment:
Since these static variables seem to only be used in one function each, they can be declared as static inside those functions, which cleans up the start of the function a bit. The Init ones stay outside functions for now.

Reply:
Sounds good, will do.

Reply:
Due to sDisplayLists, sVecZero1 and sColorBlack can't be moved since doing so would cause the data section to become non-matching. I have moved sVecZero2 to the relevant block though.

Reply:
can't sDisplayLists also be moved in?

Reply:
Uhhh.... that would be the obvious solution, wouldn't it? 😅